### PR TITLE
integrated verify cart page, and did the warning labels

### PR DESCRIPTION
**Pull request by CrazEpic**
Verify cart page now calls the backend instead of being hardcoded. Also warning checks are created. For the category rule, I interpret it as 1 unique item / category.

edit: 1 item / category after deals now

Shown are various carts. There is a 3-for-1 bagel deal.

I used Postman to call backend APIs. Ping me if you need them.

image
image
image
image

### CODE REVIEW

**Comment by CrazEpic** [comment]
image
image

File: utils/cartCalculations.ts [hunk 1]
```diff
@@ -26,13 +26,17 @@ export const pendingCartWarnings = (cart) => {
 	if (cart.CartItems.filter((cartItem) => cartItem.expiredCount > 0).length > 0) {
 		warnings.push("Cart has expired items")
 	}
-	const categories: Array<string> = []
+	const categories: { [key: string]: number } = {}
 	for (let i = 0; i < cart.CartItems.length; i++) {
-		if (categories.find((category) => category == cart.CartItems[i].Item.categoryName)) {
+		if (!categories[cart.CartItems[i].categoryName]) {
```
Comment (emoryjgrubbs) [inline utils/cartCalculations.ts:31]: "cartItems[i].category" name is undefined, replace with "cart.CartItems[i].Item.categoryName"

**Comment by emoryjgrubbs** [comment]
i'm seeing a warning for over category limit, despite the two items being in separate categories
image
see my inline comment on utils/cartCalculations for the above issue

I think there should be a mechanism to cull carts stale carts (one which are past when they would reasonably still be relevant)
image
the above image is what i'm thinking about for the search, although we could also have some api that is run at some interval to remove carts older than a cut off.

i know this is just the verify side, but i think we should make an issue for updating the cart date on submission for verification. so that we can track when a cart was submitted, displaying carts based on that date.

i would hide the count when no cart is selected

**Comment by CrazEpic** [comment]
image
image

**Comment by CrazEpic** [comment]
I'll make a separate issue for the old cart clearing

**Review by emoryjgrubbs — APPROVED** [review]
looks good

**Pull request merged**